command/plugins.go

```diff
@@ -53,19 +54,30 @@ func (r *multiVersionProviderResolver) ResolveProviders(
 				continue
 			}
 			if !reqd[name].AcceptsSHA256(digest) {
-				// This generic error message is intended to avoid troubling
-				// users with implementation details. The main useful point
-				// here is that they need to run "terraform init" to
-				// fix this, which is covered by the UI code reporting these
-				// error messages.
-				errs = append(errs, fmt.Errorf("provider.%s: installed but not yet initialized", name))
+				errs = append(errs, fmt.Errorf("provider.%s: checksum mismatch", name))
```

Comment:
This feels a bit too implementation-details for my liking. I know what we had before was a bit too far in the _other_ direction, but maybe we could compromise with something like:

```
provider.aws: new or changed plugin executable
```

The fact that we detected the change with a checksum isn't important, I think.